src/main/java/com/internet/shop/dao/jdbc/ShoppingCardDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+        String query = "INSERT INTO shopping_cart(fk_user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, cart.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+
+            if (resultSet.next()) {
+                cart.setId(resultSet.getLong("GENERATED_KEY"));
+            }
+            return cart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creating cart"
+                    + cart + " is failed", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT sc.id, sc.fk_user_id, name, price FROM products p  "
+                + "JOIN shopping_cart_products scp ON p.id = scp.product_id "
+                + "RIGHT JOIN shopping_cart sc  ON sc.id = scp.cart_id WHERE scp.cart_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+
+            List<Product> products = new ArrayList<>();
+            ShoppingCart shoppingCart = null;
+
+            while (rs.next()) {
+                if (rs.isLast()) {
+                    shoppingCart = getCart(rs);
+                    shoppingCart.setProducts(products);
+                }
+                products.add(new Product(rs.getLong("id"),
+                        rs.getString("name"), rs.getDouble("price")));
+            }
+
+            return Optional.ofNullable(shoppingCart);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting cart by id="
+                    + id + " is failed", e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        String query = "SELECT * FROM shopping_cart WHERE deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                shoppingCarts.add(getCart(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get carts from DB", e);
+        }
+        for (ShoppingCart shoppingCart : shoppingCarts) {
+            shoppingCart.setProducts(getProducts(shoppingCart.getId()));
+        }
+        return shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        throw new UnsupportedOperationException();
```

Comment:
?
in update operation just call delete products from cart and then add products to cart

Reply:
in this case we will have a lot  of redundant queries and because of this we will wast a lot of resource